dd-sdk-android/src/main/kotlin/com/datadog/android/rum/internal/tracking/ViewLoadingTimer.kt

```diff
@@ -0,0 +1,74 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.rum.internal.tracking
+
+import java.util.WeakHashMap
+
+internal class ViewLoadingTimer {
+    private val viewsTimeAndState = WeakHashMap<Any, TimeAndState>()
+
+    fun onCreated(view: Any) {
+        viewsTimeAndState[view] = TimeAndState(System.nanoTime())
+    }
+
+    fun onStartLoading(view: Any) {
+        viewsTimeAndState[view]?.let {
+            it.loadingStart = System.nanoTime()
+        }
+    }
+
+    fun onFinishedLoading(view: Any) {
+        viewsTimeAndState[view]?.let {
+            val loadingStart = it.loadingStart
+            it.loadingTime =
+                if (loadingStart != null) {
+                    System.nanoTime() - loadingStart
+                } else {
+                    // in case the view was hidden but it was resumed directly
+                    // without being started again
+                    0
+                }
+            if (it.finishedLoadingOnce) {
+                it.firstTimeLoading = false
+            }
+
+            it.loadingStart = null
+            it.finishedLoadingOnce = true
+        }
+    }
+
+    fun onViewDestroyed(view: Any) {
+        viewsTimeAndState.remove(view)
+    }
+
+    fun onViewHidden(view: Any) {
+        viewsTimeAndState[view]?.let {
+            it.loadingTime = 0
+            it.loadingStart = null
+            it.firstTimeLoading = false
+        }
+    }
+
+    fun loadingTime(view: Any): Long? {
+        viewsTimeAndState[view]?.let {
+            return it.loadingTime
+        }
+
+        return null
+    }
+
+    fun firstTimeLoading(view: Any): Boolean {
+        return viewsTimeAndState[view]?.firstTimeLoading ?: false
+    }
+
+    private data class TimeAndState(
```

Comment:
`TimeAndState` seems a bit weak. Maybe we could rename this to `ViewLoadingInfo`?